tests/datatype/tests_Base64Binary.cpp

```diff
@@ -0,0 +1,105 @@
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+
+#include <doctest/doctest.h>
+#include <rdf4cpp/rdf.hpp>
+
+using namespace rdf4cpp::rdf::datatypes::xsd;
+
+// ascii -> value
+// note: modified to decode '=' to 0
+static constexpr std::array<int8_t, 128> decode_lut{-1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,
+                                                    -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,
+                                                    -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  62,  -1,  -1,  -1,  63,
+                                                    52,  53,  54,  55,  56,  57,  58,  59,  60,  61,  -1,  -1,  -1,   0,  -1,  -1,
+                                                    -1,   0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,  14,
+                                                    15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,  -1,  -1,  -1,  -1,  -1,
+                                                    -1,  26,  27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,  39,  40,
+                                                    41,  42,  43,  44,  45,  46,  47,  48,  49,  50,  51,  -1,  -1,  -1,  -1,  -1};
+
+static std::byte base64_decode_single(char const c) {
+    auto const decoded = decode_lut[static_cast<size_t>(c)];
+    REQUIRE(decoded >= 0);
+    return static_cast<std::byte>(decoded);
+}
+
+TEST_SUITE("base64Binary") {
+    TEST_CASE("canonical") {
```

Comment:
Is canonical the right name for the testcases? some seem quite non-canonical

Reply:
canonicalization probbably fits better